Gems/ROS2/Code/Source/RobotImporter/URDF/PrefabMakerUtils.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RobotImporter/URDF/PrefabMakerUtils.h"
+#include "RobotImporter/Utils/TypeConversions.h"
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzCore/std/string/conversions.h>
+#include <AzToolsFramework/API/EditorAssetSystemAPI.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/ToolsComponents/TransformComponent.h>
+#include <Source/EditorColliderComponent.h>
+#include <Source/EditorShapeColliderComponent.h>
+
+namespace ROS2::PrefabMakerUtils
+{
+    AZ::IO::Path GetAzModelAssetPathFromModelPath(const AZ::IO::Path& modelPath)
+    {
+        bool assetFound = false;
+        AZ::Data::AssetInfo assetInfo;
+        AZ::IO::Path watchDir;
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            assetFound,
+            &AzToolsFramework::AssetSystem::AssetSystemRequest::GetSourceInfoBySourcePath,
+            modelPath.c_str(),
+            assetInfo,
+            watchDir.Native());
+
+        if (!assetFound)
+        {
+            AZ_Error("PrefabMakerUtils", false, "Could not find model asset for %s", modelPath.c_str());
+            return {};
+        }
+
+        auto assetPath = AZ::IO::Path(assetInfo.m_relativePath).ReplaceExtension("azmodel");
+        AZStd::to_lower(assetPath.Native().begin(), assetPath.Native().end());
+
+        return assetPath;
+    }
+
+    void SetEntityTransformLocal(const urdf::Pose& origin, AZ::EntityId entityId)
+    {
+        urdf::Vector3 urdfPosition = origin.position;
+        urdf::Rotation urdfRotation = origin.rotation;
+        AZ::Quaternion azRotation = URDF::TypeConversions::ConvertQuaternion(urdfRotation);
+        AZ::Vector3 azPosition = URDF::TypeConversions::ConvertVector3(urdfPosition);
+        AZ::Transform tf(azPosition, azRotation, 1.0f);
+
+        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        auto* transformInterface = entity->FindComponent<AzToolsFramework::Components::TransformComponent>();
+
+        if (!transformInterface)
+        {
+            AZ_Error("SetEntityTransform", false, "Missing Transform component!");
+            return;
+        }
+        // Alternative - but requires an active component. We would activate/deactivate a lot. Or - do it in a second pass.
+        // AZ::TransformBus::Event(entityId, &AZ::TransformBus::Events::SetLocalTM, transformForChild);
```

Comment:
Is this a TODO? Should remove and add a backlog issue instead

Reply:
done